packages/maskbook/webpack.config.ts

```diff
@@ -365,22 +367,54 @@ export default async function (cli_env: Record<string, boolean> = {}, argv: Argv
         if (target.iOS || target.runtimeEnv.target === 'firefox') return path
         return [src('./miscs/browser-loader.js'), ...path]
     }
-    function getManifestPlugin() {
-        const manifest = require('./src/manifest.json')
-        if (target.Chromium) modifiers.chromium(manifest)
-        else if (target.Firefox) modifiers.firefox(manifest)
-        else if (target.Android) modifiers.geckoview(manifest)
-        else if (target.iOS) modifiers.safari(manifest)
+    function emitManifestFile() {
+        return emitFile({
+            name: 'manifest.json',
+            content() {
+                const manifest = require('./src/manifest.json')
+                if (target.Chromium) modifiers.chromium(manifest)
+                else if (target.Firefox) modifiers.firefox(manifest)
+                else if (target.Android) modifiers.geckoview(manifest)
+                else if (target.iOS) modifiers.safari(manifest)
 
-        if (mode === 'development') modifiers.development(manifest)
-        else modifiers.production(manifest)
+                if (mode === 'development') modifiers.development(manifest)
+                else modifiers.production(manifest)
 
-        if (isManifestV3) modifiers.manifestV3(manifest)
+                if (isManifestV3) modifiers.manifestV3(manifest)
 
-        if (target.runtimeEnv.build === 'beta') modifiers.beta(manifest)
-        else if (target.runtimeEnv.build === 'insider') modifiers.nightly(manifest)
+                if (target.runtimeEnv.build === 'beta') modifiers.beta(manifest)
+                else if (target.runtimeEnv.build === 'insider') modifiers.nightly(manifest)
 
-        return new ManifestPlugin({ config: { base: manifest } })
+                return JSON.stringify(manifest, null, 4)
+            },
+        })
+    }
+    function emitManifestWebExtShimFile() {
```

Comment:
@Jack-Works replace method

Reply:
but this file is purely static. it's better to maintain if it is static.